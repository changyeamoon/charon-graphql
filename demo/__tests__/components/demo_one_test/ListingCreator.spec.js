import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { render, wait, cleanup } from 'react-testing-library';
import toJSON from 'enzyme-to-json';
import ListingCreator from '../../../client/components/demo_one/ListingCreator';
import {
  getAuthorsQuery,
  addListingMutation,
  getListingsQuery,
} from '../../../client/queries/queries';

describe('ListingCreator', () => {
  let wrapper;
  const mocks = [
    {
      request: {
        query: getAuthorsQuery,
      },
      result: {
        data: {
          authors: { id: '1', name: 's' },
        },
      },
    },
  ];
  // beforeEach(() => {
  //   wrapper = shallow(
  //     <MockedProvider mocks={mocks} addTypename={false}>
  //       <ListingCreator />
  //     </MockedProvider>,
  //   );
  // });

  // afterEach(() => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  test('should render loading state initially', () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ListingCreator />
      </MockedProvider>,
    );
    expect(container).toMatchSnapshot();
    // this way does work with enzyme mount, but could get data to render
    // expect(toJSON(wrapper.find('ListingCreator'))).toMatchSnapshot();
  });

  test('renders data', async () => {
    const { queryByText, container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ListingCreator />
      </MockedProvider>,
    );
    console.log(queryByText);
    await waitUntilLoadingIsFinished(queryByText);

    expect(container).toMatchSnapshot();
  });

  // test.skip('should render authors', () => {
  //   // console.log(wrapper.debug());
  //   wrapper.find('select').simulate('change', { target: { value: 'chang' } });
  //   console.log(wrapper);
  //   expect(
  //     wrapper
  //       .find('select')
  //       .props()
  //       .value(),
  //   ).toBe('chang');
  // });
});
