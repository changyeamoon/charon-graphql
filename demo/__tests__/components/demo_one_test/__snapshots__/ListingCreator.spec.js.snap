// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingCreator should render loading state initially 1`] = `
<div>
  <div
    class="listing-creator"
  >
    <div>
      <h3>
        Total Listings:
      </h3>
       
      <span>
        0
      </span>
    </div>
    <form>
      <div
        class="field"
      >
        <label>
          Title: 
        </label>
        <input
          name="title"
          placeholder="Enter Title"
          type="text"
          value=""
        />
        <span>
          ❌
        </span>
      </div>
      <div
        class="field"
      >
        <label>
          Author: 
        </label>
        <select>
          <option>
            Select Author
          </option>
          <option
            disabled=""
          >
            loading authors...
          </option>
        </select>
        <span>
          ❌
        </span>
      </div>
      <input
        type="submit"
        value="+"
      />
    </form>
  </div>
</div>
`;
